### DISCUSSION TITLE
Viedoconference freezes and pixelates

### DISCUSSION THREAD

**Discussion by Eirikr70** [coturn/coturn#1485] [Q&A]
Hello,
I don't much know where to start from to diagnose frequent freezing/pixelating in my videoconferences.
I have a server with Synapse and Coturn on an Odroid H4+. The TCP connexions are served by an Nginx reverse-proxy on an Rpi4B. They both are routed through ethernet to my residential router with ports 49152-49272 open to UDP traffic.
I connect to the calls with my Android phone on WiFi at 3 m distance from the WiFi router.
The measured real downstream rate is 48 MBps and the upgoing is 24.
Can anyone give me a hint where I should start for diagnosing my connexion problem ?
Thanks

**Comment by jonesmz** [comment]
You absolutely, 100%, do not want to use TCP connections or a reverse proxy via nginx to relay video connections. The head of line blocking and other problems with TCP will cause video problems.

I assume you're using webrtc? you should probably start with investigating why coturn is part of your webrtc connection, instead of being able to do a direct connection with the peer. Most likely you can eliminate the turnserver from the connection entirely.
